x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/index.ts

```diff
@@ -150,62 +130,10 @@ export class AIAssistantKnowledgeBaseDataClient extends AIAssistantDataClient {
   };
 
   /**
-   * Deploy the ELSER model with default configuration
-   */
-  private deployModel = async () => {
-    const elserId = await this.options.getElserId();
-    this.options.logger.debug(`Deploying ELSER model '${elserId}'...`);
-    try {
-      const esClient = await this.options.elasticsearchClientPromise;
-      await esClient.ml.startTrainedModelDeployment({
-        model_id: elserId,
-        wait_for: 'fully_allocated',
-      });
-    } catch (error) {
-      this.options.logger.error(`Error deploying ELSER model '${elserId}':\n${error}`);
-      throw new Error(`Error deploying ELSER model '${elserId}':\n${error}`);
-    }
-  };
-
-  /**
-   * Checks if the provided model is deployed and allocated in Elasticsearch
+   * Checks if the inference endpoint is deployed and allocated in Elasticsearch
    *
    * @returns Promise<boolean> indicating whether the model is deployed
    */
-  public isModelDeployed = async (): Promise<boolean> => {
-    const elserId = await this.options.getElserId();
-    this.options.logger.debug(`Checking if ELSER model '${elserId}' is deployed...`);
-
-    try {
-      if (this.isV2KnowledgeBaseEnabled) {
-        return await this.isInferenceEndpointExists();
-      } else {
-        const esClient = await this.options.elasticsearchClientPromise;
-        const getResponse = await esClient.ml.getTrainedModelsStats({
-          model_id: elserId,
-        });
-
-        // For standardized way of checking deployment status see: https://github.com/elastic/elasticsearch/issues/106986
-        const isReadyESS = (stats: MlTrainedModelStats) =>
-          stats.deployment_stats?.state === 'started' &&
-          stats.deployment_stats?.allocation_status.state === 'fully_allocated';
-
-        const isReadyServerless = (stats: MlTrainedModelStats) =>
-          (stats.deployment_stats?.nodes as unknown as MlTrainedModelDeploymentNodesStats[])?.some(
-            (node) => node.routing_state.routing_state === 'started'
-          );
-
-        return getResponse.trained_model_stats?.some(
```

Comment:
Before @patrykkopycinski went on PTO, we talked about this part of code and he mentioned that we still need to do these checks for V2 version as well - which was not done previously and that was missing for V2.

The actual check would be the code from the V1 path + inference endpoint existence check. Something like:

```
...
const inferenceExists = await this.isInferenceEndpointExists();
return getResponse.trained_model_stats?.some(
          (stats) => isReadyESS(stats) || isReadyServerless(stats)
) && inferenceExists;
```

Reply:
This should take care of it! https://github.com/elastic/kibana/pull/198180/commits/499454406b287157d5a3c0e373c0444d047cdd59